src/rp2_common/hardware_clocks/clocks.c

```diff
@@ -327,10 +327,14 @@ void clock_gpio_init(uint gpio, uint src, uint div) {
         invalid_params_if(CLOCKS, true);
     }
 
+    // set up div
+    invalid_params_if(CLOCKS, div <= 0);
+    uint udiv = div * (1u << 8u);
+
     // Set up the gpclk generator
     clocks_hw->clk[gpclk].ctrl = (src << CLOCKS_CLK_GPOUT0_CTRL_AUXSRC_LSB) |
                                  CLOCKS_CLK_GPOUT0_CTRL_ENABLE_BITS;
-    clocks_hw->clk[gpclk].div = div << CLOCKS_CLK_GPOUT0_DIV_INT_LSB;
+    clocks_hw->clk[gpclk].div = udiv;
```

Comment:
_Might_ be worth modifying this code to match the style of e.g. https://github.com/raspberrypi/pico-sdk/blob/develop/src/rp2_common/hardware_adc/include/hardware/adc.h#L164 ?